pkgs/development/interpreters/cling/default.nix

```diff
@@ -0,0 +1,101 @@
+{ clangStdenv,
+  python,
+  fetchFromGitHub,
+  libffi,
+  git,
+  cmake,
+  zlib,
+  fetchgit,
+  makeWrapper,
+  runCommand,
+  llvmPackages_5,
+  callPackage,
+
+  # Provide flags to Cling to prevent it from looking for system include paths,
+  # and pass it the paths to glibc and the LLVM C++
+  isolate ? true,
+
+  # Produce a "standalone" version of Cling, suitable for running as an executable.
+  # If set to false, the output will include extra files for using Cling as a library.
+  standalone ? false
+}:
+
+let
+  # For using cling as a library (i.e. with xeus-cling)
+  clingFull = clangStdenv.mkDerivation rec {
```

Comment:
Do you really need clangStdenv here? It should correspond to the current version of `clang` which is clang 7. Is that okay?

Reply:
I'm still thinking about this. It's related to the conversation with @Axel-Naumann about which C++ standard lib to link against. I *think* building with Clang 7 is fine but need to experiment a bit more.